Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -144,6 +146,20 @@ namespace AzToolsFramework
             GarbageCollectTemplates(false);
         }
 
+        void PrefabSystemComponent::OnSourceFilePathNameChanged(const AZStd::string_view fromPathName, const AZStd::string_view toPathName)
+        {
+            // Convert to prefab relative path
+            AZ::IO::Path fromRelativePath = m_prefabLoader.GenerateRelativePath(fromPathName);
+
+            // Update loaded template with new pathname
+            if (Prefab::TemplateId templateId = GetTemplateIdFromFilePath(fromRelativePath); templateId != InvalidTemplateId)
```

Comment:
We are just supporting file renames and moves within Asset Browser.

Reply:
I know, but the API sending this ebus event has a flag `isFolder`, so while I haven't tested it (I know APB supports it at least) I assume a whole folder containing many prefabs can be moved/renamed.  If that's the case, it seems like the event would broadcast that the folder was renamed, then `GetTemplateIdFromFilePath` would likely return nothing (since the folder isn't a prefab) and end up exiting out.

Reply:
Oh I actually didn't realize users can rename/move folders in AB. I'll update to handle this case. Thanks!